src/Elastic.Apm.Azure.ServiceBus/MicrosoftAzureServiceBusDiagnosticListener.cs

```diff
@@ -0,0 +1,240 @@
+					break;
+			}
+		}
+
+		private void OnReceiveStart(KeyValuePair<string, object> kv, string action, PropertyFetcherCollection cachedProperties)
+		{
+			if (kv.Value is null)
+			{
+				Logger.Trace()?.Log("Value is null - exiting");
+				return;
+			}
+
+			var queueName = cachedProperties.Fetch(kv.Value,"Entity") as string;
+			if (MatchesIgnoreMessageQueues(queueName))
+				return;
+
+			var transactionName = queueName is null
+				? $"{ServiceBus.SegmentName} {action}"
+				: $"{ServiceBus.SegmentName} {action} from {queueName}";
+
+			var transaction = ApmAgent.Tracer.StartTransaction(transactionName, ServiceBus.Type);
+			transaction.Context.Service = _service;
+
+			// transaction creation will create an activity, so use this as the key.
+			var activityId = Activity.Current.Id;
+
+			if (!_processingSegments.TryAdd(activityId, transaction))
+			{
+				Logger.Trace()?.Log(
+					"Could not add {Action} transaction {TransactionId} for activity {ActivityId} to tracked segments",
+					action,
+					transaction.Id,
+					activityId);
+			}
+		}
+
+		private bool MatchesIgnoreMessageQueues(string name)
+		{
+			if (name != null && _realAgent != null)
+			{
+				var matcher = WildcardMatcher.AnyMatch(_realAgent.ConfigStore.CurrentSnapshot.IgnoreMessageQueues, name);
+				if (matcher != null)
+				{
+					Logger.Debug()?.Log(
+						"Not tracing message from {QueueName} because it matched IgnoreMessageQueues pattern {Matcher}",
+						name,
+						matcher.GetMatcher());
+					return true;
+				}
+			}
+
+			return false;
+		}
+
+		private void OnSendStart(KeyValuePair<string, object> kv, string action, PropertyFetcherCollection cachedProperties)
+		{
+			var currentSegment = ApmAgent.GetCurrentExecutionSegment();
+			if (currentSegment is null)
+			{
+				Logger.Trace()?.Log("No current transaction or span - exiting");
+				return;
+			}
+
+			if (kv.Value is null)
+			{
+				Logger.Trace()?.Log("Value is null - exiting");
+				return;
+			}
+
+			var activity = Activity.Current;
+			var queueName = cachedProperties.Fetch(kv.Value,"Entity") as string;
+			var destinationAddress = cachedProperties.Fetch(kv.Value, "Endpoint") as Uri;
+
+			if (MatchesIgnoreMessageQueues(queueName))
+				return;
+
+			var spanName = queueName is null
+				? $"{ServiceBus.SegmentName} {action}"
+				: $"{ServiceBus.SegmentName} {action} to {queueName}";
+
+			var span = currentSegment.StartSpan(spanName, ServiceBus.Type, ServiceBus.SubType, action.ToLowerInvariant());
+
+			span.Context.Destination = new Destination
+			{
+				Address = destinationAddress?.AbsoluteUri,
+				Service = new Destination.DestinationService
+				{
+					Name = ServiceBus.SubType,
+					Resource = queueName is null ? ServiceBus.SubType : $"{ServiceBus.SubType}/{queueName}",
+					Type = ServiceBus.Type
+				}
+			};
+
+			if (!_processingSegments.TryAdd(activity.Id, span))
+			{
+				Logger.Trace()?.Log(

```

Comment:
Similarly to above, here we should do something with the span we just created.

Reply:
See https://github.com/elastic/apm-agent-dotnet/pull/1225#discussion_r603722336